src/vs/workbench/services/editor/common/editorGroupsService.ts

```diff
@@ -390,6 +390,11 @@ export interface IEditorGroup {
 	 */
 	readonly onWillCloseEditor: Event<IEditorCloseEvent>;
 
+	/**
+	 * An event that is fired when an editor is about to move
+	 */
+	readonly onWillMoveEditor: Event<IEditorInput>;
```

Comment:
I can make the event better. I really just formed the event around the Notebook hack they were using to spy on the tab so that it could no longer depend on an open override with a context. Let me fortify the event some more and make it look less hacky

Reply:
Yeah sounds good, to clarify I think the event is good and not a "hack" but it might miss a few properties.

Reply:
Alright I have made this event a bit more robust :)